services/masterdata/src/lib/components/masterdata/alternative-keys/alternative-key-list/alternative-key-list.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, Input, OnInit, OnDestroy } from "@angular/core";
+import { TranslateService } from "@ngx-translate/core";
+import { Subscription } from "rxjs/internal/Subscription";
+import { ILazyLoadEvent } from "@lens/ui-prime-components";
+import { MasterdataAlternativeKey } from "../../../../core/models";
+import { MasterdataCrudHttpService } from "../../../../core/services";
+import { MasterdataAlternativeKeyService } from "../alternative-key.service";
+
+@Component({
+	selector: "masterdata-alternative-key-list",
+	templateUrl: "./alternative-key-list.component.html"
+})
+export class MasterdataAlternativeKeyListComponent implements OnInit, OnDestroy {
+	isLoading = false;
+	items: MasterdataAlternativeKey[] = [];
+	totalSize = 0;
+	alternativeKeyAddedSubscription: Subscription;
+
+	@Input() public typeId = "";
+	@Input() public masterdataId = "";
+
+	constructor(
+		private readonly service: MasterdataCrudHttpService,
+		private readonly translateService: TranslateService,
+		private readonly alternativeKeyService: MasterdataAlternativeKeyService
+	) {
+		this.alternativeKeyAddedSubscription = this.alternativeKeyService.alternativeKeyAdded$.subscribe(key => {
+			this.loadItems(0, 0);
+		});
+	}
+
+	ngOnInit(): void {
+		this.isLoading = true;
+	}
+
+	ngOnDestroy() {
+		if (this.alternativeKeyAddedSubscription) {
+			this.alternativeKeyAddedSubscription.unsubscribe();
+		}
+	}
+
+	loadItems(offset: number, rows: number) {
+		this.isLoading = true;
+		this.service.getAllMasterdataAlternativeKeys(this.typeId, this.masterdataId, offset, rows).subscribe({
+			next: data => {
+				this.items = data.value || [];
+				this.totalSize = data.totalSize || 0;
+				this.isLoading = false;
+			},
+			complete: () => {
```

Comment:
dont forget the onerror: () => this.isLoading = false

tip: doublecheck every subscribe method, so you never run into a disabled form when the api returns an non ok response